spyder/plugins/editor/plugin.py

```diff
@@ -2339,6 +2335,31 @@ def current_file_changed(self, filename, position):
             pdb_last_step = self.main.ipyconsole.get_pdb_last_step()
             self.update_pdb_state(pdb_state, pdb_last_step)
 
+    def current_editor_cursor_changed(self, line, column):
+        """Handles the change of the cursor inside the current editor."""
+        code_editor = self.get_current_editor()
+        filename = code_editor.filename
+        position = code_editor.get_position('cursor')
+        self.add_cursor_position_to_history(to_text_string(filename), position,
+                                            fc=True)
+
+    def will_change_file(self):

```

Comment:
This method is no longer necessary if you change things according to my long comment above.